File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int currMin = Integer.MAX_VALUE, currMaxProfit = 0;
        
        for (Integer price: prices) {
            if (price < currMin) {
                currMin = price;
            }
            else if ((price - currMin) > currMaxProfit) {
                currMaxProfit = price - currMin;
            }
        } //for
        return currMaxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = 2**32 - 1
        maxProfit = 0
        for p in prices:
            if p < minPrice:
                minPrice = p
            elif maxProfit < p - minPrice:
                maxProfit = p - minPrice
        return maxProfit